File: vertx-lang-php-jphp/src/main/php/io/vertx/php/core/http/RequestOptions.php
<?php 
namespace io\vertx\php\core\http;

class RequestOptions  
{ 
    public function __construct($requestOptions)
    { 
        
    }
    
    /**
     * @return String 
     */
    public function getHost()
    { 
        return "";
    }
    
    /**
     * @param $host String 
     * @return $this
     */
    public function setHost($host)
    { 
        return $this;
    }
    
    /**
     * @return int 
     */
    public function getPort()
    { 
        return 1;
    }
    
    /**
     * @param $port int 
     * @return $this
     */
    public function setPort($port)
    { 
        return $this;
    }
    
    /**
     * @return bool 
     */
    public function isSsl()
    { 
        return false;
    }
    
    /**
     * @param $ssl bool 
     * @return $this
     */
    public function setSsl($ssl)
    { 
        return $this;
    }
    
    /**
     * @return String 
     */
    public function getURI()
    { 
        return "";
    }
    
    /**
     * @param $uri String 
     * @return $this
     */
    public function setURI($uri)
    { 
        return $this;
    }
}

File: vertx-lang-php-jphp/src/main/data/io/vertx/php/core/http/RequestOptions.java
package io.vertx.php.core.http;

import io.vertx.core.json.JsonObject;
import io.vertx.lang.jphp.wrapper.DataObjectWrapper;
import org.develnext.jphp.zend.ext.json.JsonFunctions;
import php.runtime.Memory;
import php.runtime.memory.ObjectMemory;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;
import php.runtime.annotation.Reflection.*;

@SuppressWarnings("unused")
@Name("RequestOptions")
@Namespace("io\\vertx\\php\\core\\http")
//false
public  class RequestOptions extends DataObjectWrapper<io.vertx.core.http.RequestOptions> { 
    public RequestOptions(Environment env, io.vertx.core.http.RequestOptions wrappedObject) {
        super(env, wrappedObject);
    }

    public RequestOptions(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Signature
    public Memory __construct(){
        this.__wrappedObject = new io.vertx.core.http.RequestOptions();
        return Memory.NULL;    }

    @Signature
    public Memory __construct(Memory memory) {
        this.__wrappedObject = new io.vertx.core.http.RequestOptions(new JsonObject(JsonFunctions.json_encode(memory)));
        return Memory.NULL;
    }

    @Signature
    public String getHost(){ 
        return this.getWrappedObject().getHost();
    }

    @Signature
    public RequestOptions setHost(String host){ 
        this.getWrappedObject().setHost(host);
        return this;
    }

    @Signature
    public int getPort(){ 
        return this.getWrappedObject().getPort();
    }

    @Signature
    public RequestOptions setPort(int port){ 
        this.getWrappedObject().setPort(port);
        return this;
    }

    @Signature
    public boolean isSsl(){ 
        return this.getWrappedObject().isSsl();
    }

    @Signature
    public RequestOptions setSsl(boolean ssl){ 
        this.getWrappedObject().setSsl(ssl);
        return this;
    }

    @Signature
    public String getURI(){ 
        return this.getWrappedObject().getURI();
    }

    @Signature
    public RequestOptions setURI(String uri){ 
        this.getWrappedObject().setURI(uri);
        return this;
    }
}
